### DISCUSSION TITLE
$reset

### DISCUSSION THREAD

**Discussion by KevinBatdorf** [alpine-collective/alpine-magic-helpers#11] [General]
This could be useful to reset the entire state by default, or optionally pass in specific properties.

```html
<div x-data="{ name: '', email: '', anotherProperty: ''  }">
  <!-- inside here would be a form -->
  <button @click="$reset(['name', 'email'])">Reset form</button>
</div>
```

Maybe as easy as adding the initial state as a property to `$el`?

**Comment by SimoTod** [comment]
We already have saferEval so maybe we can just read x-data, evaluate it and set the result to the component?
The version accepting properties requires a bit more work but we can probably calculate the diff as you do with track.

**Reply by KevinBatdorf** [reply]
Or maybe using saferEval and plucking out the specific data? Could also probably just save the initial state using WeakMap too.
